### PR TITLE
Offspring STI Model

### PR DESCRIPTION
**Pull request by sebjimenez**
# Purpose or Intent

- The goal of this PR is to create  STI **model** for **offspring**

### CODE REVIEW

**Comment by sergio-ocon** [comment]
Chapter 9.4 of the "Rails 4 way" talks about STI

**Comment by sebjimenez** [comment]
All tests have passed. Please review and provide feedback. @sergio-ocon @amaurygonzalez

File: app/models/offspring.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Offspring < ApplicationRecord
+  belongs_to :user
+  self.inheritance_column = :type
+
+  class << self
+    def types
+      %w(AgedOffspring GradedOffspring)
```
Comment (sergio-ocon) [inline app/models/offspring.rb]: Why are you doing this? You don't need to define this here. In fact, that is against inheritance, as you would need to change this for another class.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class User < ApplicationRecord
+  has_many :offsprings
```
Comment (sergio-ocon) [inline app/models/user.rb]: You need to destroy the offsprings when the parent is destroyed

File: db/migrate/20160921115843_create_offsprings.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateOffsprings < ActiveRecord::Migration[5.0]
+  def change
+    create_table :offsprings do |t|
```
Comment (sergio-ocon) [inline db/migrate/20160921115843_create_offsprings.rb]: No, you don't create different tables, you create a table with everything you need

File: db/migrate/20160921120401_create_aged_offsprings.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class CreateAgedOffsprings < ActiveRecord::Migration[5.0]
+  def change
+    create_table :aged_offsprings do |t|
+      t.integer :age
```
Comment (sergio-ocon) [inline db/migrate/20160921120401_create_aged_offsprings.rb]: This goes into offsprings

File: app/models/offspring.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class Offspring < ApplicationRecord
+  belongs_to :user
+  self.inheritance_column = :type
```
Comment (sergio-ocon) [inline app/models/offspring.rb]: You don't need to do this, Rails knows you are doing this if a column is type

File: db/schema.rb [hunk 6]
```diff
@@ -31,6 +31,18 @@
     t.index ["reset_password_token"], name: "index_admins_on_reset_password_token", unique: true
   end
 
+  create_table "aged_offsprings", force: :cascade do |t|
```
Comment (sergio-ocon) [inline db/schema.rb]: Just you don't do this. It is call "single table" for a reason

File: db/schema.rb [hunk 7]
```diff
@@ -39,6 +51,14 @@
     t.index ["singleton_guard"], name: "index_myconfigs_on_singleton_guard", unique: true
   end
 
+  create_table "offsprings", force: :cascade do |t|
+    t.string   "first_name", null: false
+    t.string   "last_name"
+    t.string   "type"
+    t.datetime "created_at", null: false
```
Comment (sergio-ocon) [inline db/schema.rb]: No, you need to add age and grade here

File: spec/models/aged_offspring_spec.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe AgedOffspring, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (sergio-ocon) [inline spec/models/aged_offspring_spec.rb]: Use TDD; create the specs first

File: spec/models/graded_offspring_spec.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe GradedOffspring, type: :model do
```
Comment (sergio-ocon) [inline spec/models/graded_offspring_spec.rb]: No merge without specs

File: spec/models/offspring_spec.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Offspring, type: :model do
```
Comment (sergio-ocon) [inline spec/models/offspring_spec.rb]: No merge without specs

**Pull request closed without merging**